### PR TITLE
add stack charts plotter for metric groups

### PR DESCRIPTION
**Pull request by NiuJ1ao**
- plotter detects any metric groups in the .csv file and plot stack charts for them.

For example, if a CSV file has _Frontend_Bound_, _Bad_Speculation_, _Backend_Bound_ and _Retiring_, The plotter plots a stack chart for _Level_1_ (level 1 is the parent of four nodes) as following:
Level_1

Signed-off-by: NiuJ1ao <26167136+NiuJ1ao@users.noreply.github.com>

### CODE REVIEW

File: profile/toplev_metrics.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
+{"Level_1":
```
Comment (ustiugov) [inline profile/toplev_metrics.json:1]: if this file is copied from the pmu tools repo, it should be fetched at the setup time and should not be in this repo.

File: profile/plotter.go [hunk 2]
```diff
@@ -34,17 +38,25 @@ import (
...
-func PlotCVS(xStep int, filePath, inFile, xLable string) {
+// PlotLineCharts plots every attribute as VM number increases in the csv file
+func PlotLineCharts(xStep int, filePath, inFile, xLable string) {
 	var (
 		records = readResultCSV(filePath, inFile)
 		rows    = len(records)
 		cols    = len(records[0])
 	)
 
+	if rows < 3 {
```
Comment (ustiugov) [inline profile/plotter.go:55]: why 3? add a comment

**Review by ustiugov — CHANGES_REQUESTED** [review]
please add a thorough description of the feature with an example of a plot. The code seems fine apart from few comments. the json should be fetched at the setup time unless there is a good reason for having this file in this repo.

File: profile/toplev_metrics.json (hunk 1 above)
Comment (NiuJ1ao) [reply profile/toplev_metrics.json:1]: This file is not from pmu tools repo. JSON structure would be easier to parse and check the relations between nodes.

File: profile/plotter.go (hunk 2 above)
Comment (NiuJ1ao) [reply profile/plotter.go:55]: Done

**Comment by ustiugov** [comment]
you should not put the plot into the repo unless you add it to `*md` docs (which you may want to do). For the PR's description, you can just drag-and-drop a figure/screenshot when editing.

**Comment by ustiugov** [comment]
please merge the last 2 commits. please try to avoid creating commits with the same logical value.

**Review by ustiugov — CHANGES_REQUESTED** [review]
please merge the last 2 commits

**Review by ustiugov — CHANGES_REQUESTED** [review]
please remove the figure from the docs/ folder

**Review by ustiugov — DISMISSED** [review]

**Review by ustiugov — APPROVED** [review]

**Pull request merged**